documentation/wiki/CollectedTelemetry.md

```diff
@@ -0,0 +1,96 @@
+
+Visual Studio collects some build related telemetry - but that is not leveraging any MSBuild instrumentation, but rather information about count and duration of MSBuild API invocations from the caller point of view. For general information about telemetry being collected by Visual Studio Family of products and regulations compliance please refer to [the official documentation](https://learn.microsoft.com/en-us/compliance/regulatory/gdpr-dsr-visual-studio-family).
+
+## Type of data collected
+
+To tailor modernization and performance optimization investments we need to contain *anonymized* **Usage Data**. Those reflect type of features being used and execution time being spent in them.
+
+## Opting out
+
+MSBuild telemetry collection (that is turned on by default), can be opted out - same as .NET SDK telemetry in general - via setting `DOTNET_CLI_TELEMETRY_OPTOUT` environment variable to `1` or `true`.
+
+## Datapoints overview
+
+### Logging Configuration
+
+Expressed and collected via [LoggingConfigurationTelemetry type](https://github.com/dotnet/msbuild/blob/94941d9cb26bb86045452b4a174a357b65a30c99/src/Framework/Telemetry/LoggingConfigurationTelemetry.cs)
+
+| SDK versions | Data |
+|--------------|------|
+| >= 8.0.1     | Indication if terminal logger was used. |
+| >= 8.0.1     | User choice on terminal logger enablement. |
+| >= 8.0.1     | Source of user choice on terminal logger enablement. |
+| >= 8.0.1     | Default choice on terminal logger enablement. |
+| >= 8.0.1     | Source of default choice on terminal logger enablement. |
+| >= 8.0.1     | Indication if Console logger was used. |
+| >= 8.0.1     | Console logger type (serial, parallel). |
+| >= 8.0.1     | Console logger verbosity. |
+| >= 8.0.1     | Indication if File logger was used. |
+| >= 8.0.1     | File logger type (serial, parallel). |
+| >= 8.0.1     | Number of file loggers. |
+| >= 8.0.1     | File logger verbosity. |
+| >= 8.0.1     | Indication if Binary logger was used. |
+| >= 8.0.1     | Indication if Binary logger used with default log name. |
+
+### BuildCheck
+
+Expressed and collected via [BuildCheckTelemetry type](https://github.com/dotnet/msbuild/blob/94941d9cb26bb86045452b4a174a357b65a30c99/src/Framework/Telemetry/BuildCheckTelemetry.cs)
+
+#### BuildCheck Run
+
+| SDK versions | Data |
+|--------------|------|
+| >= 9.0.1     | Corelation guid for the run |
+| >= 9.0.1     | Count of enabled rules for the run |
+| >= 9.0.1     | Count of enabled custom rules for the run |
+| >= 9.0.1     | Count of violations encountered for the run |
+| >= 9.0.1     | Execution time spent by BuildCheck infrastructure and rules |
+
+#### BuildCheck Rule in a run
+
+| SDK versions | Data |
+|--------------|------|
+| >= 9.0.1     | Corelation guid for the run. |
```

Comment:
Do we keep track of runs using a guid for each run?

Reply:
Yes - just for the buildchecks though.

A guid is generated after build run and added to each buildcheck datapoint for that run - to be able to connect those together